### ISSUE TITLE
Object.dirname is deprecated.

### ISSUE THREAD

**Issue by jpmolinamatute** [AtomLinter/linter-eslint#973]
Argument to `path.dirname` must be a string
```
Object.dirname (/usr/lib/atom/src/electron-shims.js:9:10)
<unknown> (/home/juanpa/.atom/packages/linter-eslint/src/main.js:253:28)
undefined.next (null:null:null)
step (/home/juanpa/.atom/packages/linter-eslint/src/main.js:8:191)
<unknown> (/home/juanpa/.atom/packages/linter-eslint/src/main.js:8:437)
<unknown> (/home/juanpa/.atom/packages/linter-eslint/src/main.js:8:99)
```

**Comment by Arcanemagus** [comment]
What version of Atom are you on? It should be impossible to hit this.

The only way this could happen is if the `TextEditor` doesn't have a path, but we are already bailing out if the `TextEditor` is modified... and all `TextEditor`s that aren't saved (the only way not to have a path) are marked as modified. 😕.

**Comment by jpmolinamatute** [comment]
I'm using Atom 1.18.0 
$ apm --version
(node:19881) [DEP0022] DeprecationWarning: os.tmpDir() is deprecated. Use os.tmpdir() instead.
apm  1.18.3
npm  5.3.0
node 8.2.1 x64
python 2.7.13
git 2.13.3

maybe this is related to node or upstream...

**Comment by Arcanemagus** [comment]
Duplicate of #972

Sorry for not catching that this was a duplicate earlier, and this was fixed in #978. Thanks for reporting it!
